### DISCUSSION TITLE
Firebase analytics user ID

### DISCUSSION THREAD

**Discussion by matasmaciulaitis** [invertase/react-native-firebase#6185] [Q&A] [answered]
Hello, I set my user id with `analytics().setUserId('123456')` and tested with debug view, this data is sent to _analytics_, but I can't find it anywhere in _analytics_ tab. Maybe it should be attached to all events? When doing exactly the same with _crashlytics_ everything is working fine and user id is attached to crash data.

**Comment by mikehardy** [comment]
Hi there! Have you searched previous issues / discussions? I believe this has been examined before and it was a quirk of DebugView on the back-end, with the userid showing up in regular analytics, just not DebugView - indicating data transmission was okay but something about DebugView (out of our control) wasn't working?

At the same time, I seem to recall a couple different items on the review pile still pending where we *may* be at fault with evidence of a reproduction from either quickstart-android or quickstart-ios showing that the native SDKs performed correctly but RNFB did not

**Reply by matasmaciulaitis** [reply]
Hey, I am not sure if we are on the same page. My issues is that I can't find any place to check if user id is set besides Debug View. Maybe you could provide an example/screenshot where to find it in analytics tab. Thank you :)

**Comment by matasmaciulaitis** [comment, marked as answer]
Hi again, I managed to solve it. As it seems you can't access this data in analytics tab itself you have to export it to BigQuery. My mistake was that I used BigQuery extension for Firestore collections, not the BigQuery itself and link it with analytics.

**Reply by mikehardy** [reply]
To be clear / make sure I understand: you do actually see the userid in BigQuery exports of analytics data, after setting it via react-native-firebase analytics API call setUserId ? But you still do *not* see it in the DebugView of analytics (or any other location?)

That would *imply* that react-native-firebase is doing things correctly but is still not a great experience so could use some investigation in the form of checking the quickstart's for analytics at the native firebase SDK level to see if they have different behavior (and thus isolate where this breakdown is). It's an area that could still be improved in other words, and it's still not 100% clear if it's something we can improve.

But if my understanding in the first sentence is correct that is at least *something* where you may analyze user behavior...
